test/client.js

```diff
@@ -97,6 +97,8 @@ test('basic get with query params', (t) => {
       bool: 'true',
       foo: '1',
       bar: 'bar',
+      nullVal: '',
```

Comment:
can you also add an object and a function to the mix?

Reply:
those tests are in place to document how undici itself behaves. what library or algorithm is used underneath is an implementation detail.

Reply:
That's a really good argument. I added a new commit and updated the tests.